kong/runloop/handler.lua

```diff
@@ -1041,14 +1041,14 @@ return {
 
       local trusted_ip = kong.ip.is_trusted(realip_remote_addr)
       if trusted_ip then
-        forwarded_proto = var.http_x_forwarded_proto or scheme
-        forwarded_host  = var.http_x_forwarded_host  or host
-        forwarded_port  = var.http_x_forwarded_port  or port
+        forwarded_proto = route.x_forwarded_proto or service.x_forwarded_proto or var.http_x_forwarded_proto or scheme
```

Comment:
In general these (trusted) ones should be in this order:

```suggestion
        forwarded_proto = var.http_x_forwarded_proto or route.x_forwarded_proto or service.x_forwarded_proto or scheme
```

But I feel like we use a sledgehammer to implement this. At least we should only have this on route or service but not on both. Perhaps only on service. Not sure if this should be on `db` or say ENV variable that you pass to Kong when it is started.

Reply:
This is how Amazon does it, and this works just fine with Kong:
https://kevinhakanson.com/2018-03-15-amazon-load-balancers-x-forwarded-headers-and-proxy-protocol-support

I am not sure why Docker cannot do it.

Reply:
If we use a env var for this case I see two issues:

1. It's a implicit behaviour for every registered service and will cause new issues. 
2. Maybe I want to set different values for the services. 

In my opinion my solution proposal is the cleanest possible way to allow people to configure kong as they need it

Reply:
Ah, ok I see, you want to break X-Forwarded even in case where  you trust them.